openhands/app_server/event_callback/webhook_router.py

```diff
@@ -186,3 +188,16 @@ async def _run_callbacks_in_bg_and_close(
         # We don't use asynio.gather here because callbacks must be run in sequence.
         for event in events:
             await event_callback_service.execute_callbacks(conversation_id, event)
+
+
+def _import_all_tools():
+    """We need to import all tools so that they are available for deserialization in webhooks."""
+    for _, name, is_pkg in pkgutil.walk_packages(tools.__path__, tools.__name__ + '.'):
+        if is_pkg:  # Check if it's a subpackage
+            try:
+                importlib.import_module(name)
+            except ImportError as e:
+                _logger.error(f"Warning: Could not import subpackage '{name}': {e}")
+
+
+_import_all_tools()
```

Comment:
Do we want to import everything here or whitelist specific ones? 

OK either way!